File: Asmi Bhaskar/day 21.c
#include <stdio.h>
#include <limits.h>

int minCoins(int coins[], int N, int V) {
    int dp[V + 1];
    dp[0] = 0;

    for (int i = 1; i <= V; ++i) {
        dp[i] = INT_MAX;
        for (int j = 0; j < N; ++j) {
            if (i >= coins[j] && dp[i - coins[j]] != INT_MAX) {
                dp[i] = dp[i - coins[j]] + 1;
            }
        }
    }

    return (dp[V] == INT_MAX) ? -1 : dp[V];
}

int main() {
    int coins[] = {9, 6, 5, 1};
    int N = sizeof(coins) / sizeof(coins[0]);
    int V = 11;

    int result = minCoins(coins, N, V);
    printf("Minimum number of coins required: %d\n", result);

    return 0;
}

File: Shailja/Day21.java
import java.util.Arrays;

public class Day21 {
    public static void main(String[] args) {
        int[] coins = { 9,6, 5,1 };
        int v = 11;
        int result = minCoins(coins, v);
        if (result == -1) {
            System.out.println("It's not possible to make the change.");
        } else {
            System.out.println("Minimum number of coins required: " + result);
        }
    }
    public static int minCoins(int[] coins, int v) {
        int[] dp = new int[v + 1];
        Arrays.fill(dp, Integer.MAX_VALUE);
        dp[0] = 0;

        for (int coin : coins) {
            for (int j = coin; j <= v; j++) {
                if (dp[j - coin] != Integer.MAX_VALUE) {
                    dp[j] = Math.min(dp[j], dp[j - coin] + 1);
                }
            }
        }

        return dp[v] == Integer.MAX_VALUE ? -1 : dp[v];
    }

   
}
